hw/ip/otbn/dv/otbnsim/sim/edn_client.py

```diff
@@ -64,14 +64,19 @@ def forget(self) -> None:
     def take_word(self, word: int) -> None:
         '''Take a 32-bit data word that we've been waiting for'''
         assert 0 <= word < (1 << 32)
-        assert self._acc is not None
-        assert len(self._acc) < ACC_LEN
-        assert self._cdc_counter is None
+        # Ignore incoming EDN words if there is no request for that word.
+        # We might still get words if we are in the middle of an EDN request
+        # but also had a reset (in which we drop the request flag)
```

Comment:
I suggest to integrate this comment into the docstring of `take_word` (currently on Line 65). Something along the lines of:

Take a 32-bit data word, which we requested. Two cases have to be distinguished:
- If there has not been a reset between the request and this incoming word, we append the word to the internal `_acc` list.
- If there has been a reset between the request and this incoming word, we ignore the word.

If you think the case distinction is an implementation detail that does not belong into the function docstring, I suggest to remove the "that we've been waiting for" part of the docstring (because it's misleading) and put the case distinction into a comment below the docstring.

Reply:
I think explaining it in a simple manner in docstring makes sense. I changed the wording a little bit but still applied your suggestion, thanks!